### ISSUE TITLE
We shouldn't require authors to be in config-references.yml to run a recipe

### ISSUE THREAD

**Issue by ledm** [ESMValGroup/ESMValCore#28]
In the documentation section of recipes, there is an obligatory field: `authors`. This requires the names listed in the `authors` section of the recipe to be in the `config-references.yml` file.

Otherwise, you will get an error like this one:
```
ValueError: Tag 'lee' does not exist in section 'authors' of /users/modellers/ledm/workspace/ESMValToolTest/ESMValTool/esmvaltool/config-references.yml

```

This should 100% be a requirement for pushing recipes back to github.  

However, this excludes new users from ownership of their own recipe if they don't have access to the source code. For instance they are running a shared installation or ESMValTool, or they are using the `user` installation instead of the `develop` installation.

Can we relax this requirement when running a recipe, but keep it for submitting recipes? 

@valeriupredoi and I spoke about it earlier today. Any thoughts @mattiarighi, @bouweandela ?

**Comment by bouweandela** [comment]
I took this into account when splitting the packages. @ledm After https://github.com/ESMValGroup/ESMValTool/pull/1148 is merged, can you please test if things now work as you would like?

**Comment by bouweandela** [comment]
This has now been implemented in the code, but it should still be documented.

Note that when the config-references.yml file is not available, no information from that file can be retrieved, so authors should write their full name and references etc. in the recipe and diagnostic scripts if they still want reasonable looking provenance output.

**Comment by stefsmeets** [comment]
Hi @bouweandela , has this ever been resolved? As we are expanding the API, it is getting increasingly difficult for me to test recipe / metadata rendering since this depends on the ability of author names (tags) being resolved. Part of this is the forward dependency on `esmvaltool` to define author names, and the other is that there is no 'default' user to fall back to or to test with.

**Comment by bouweandela** [comment]
No, this has not been completely solved, or the issue would have been closed. If you try to run a recipe with an author tag that is not available in the config-references.yml file (author `some_author` in this case), the run will stop with this error message:
```
ValueError: Tag 'some_author' does not exist in section 'authors' of /home/bandela/src/esmvalgroup/esmvaltool/esmvaltool/config-references.yml
```

The problem is that a collection of recipes, diagnostic scripts, bibtex references, and config-references.yml files needs to match each other. However, it is possible to use ESMValCore to just run a recipe and associated diagnostic script without the bibtex references and config-references.yml file, but obviously, you cannot replace tags with a value if you do not have the contents to replace them with, so this information will be missing from the output.

**Comment by stefsmeets** [comment]
From your previous comment I got the impression that it was resolved, just not documented. I don't just want to run a recipe, I want to make sure the resolvers work correctly for the API / html generation. The outcome depends too much on whether esmvaltool is installed or not.

I can work around the issue in the tests, but I am not super happy with the direction that is taking. I see other parts of the code have similar work-arounds. I think it would be useful to have a sort of representative 'mini-repo' inside the core which we can import from and test with (filled with john-doe's and tags / recipes / references for testing).

**Comment by stefsmeets** [comment]
Example of the issues with the tests from #991 : https://app.circleci.com/pipelines/github/ESMValGroup/ESMValCore/4261/workflows/1c532868-d1b1-4e91-8673-cf00bba300da/jobs/19824/steps

This is a run of esmvaltool in a subprocess in the tests, where we cannot patch in mocked diagnostics / tags. This problem gets exposed in #991 because we need to do this write an html summary. This means the author tags must be resolved, but they can't be, because the TAGS dict is empty I think.
